iml-services/iml-ostpool/src/db.rs

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::Error;
+use futures::future::try_join_all;
+use iml_postgres::Client;
+use iml_wire_types::{
+    db::{
+        FsRecord, ManagedHostRecord, ManagedMdtRecord, ManagedOstRecord, ManagedTargetMountRecord,
+        ManagedTargetRecord, Name, OstPoolOstsRecord, OstPoolRecord,
+    },
+    Fqdn,
+};
+use std::{collections::HashSet, sync::Arc};
+
+pub struct PoolClient {
+    client: Arc<Client>,
+    fqdn: String,
+    pool_type_id: i32,
+}
+
+impl Clone for PoolClient {
+    fn clone(&self) -> Self {
+        PoolClient {
+            client: Arc::clone(&self.client),
+            fqdn: self.fqdn.clone(),
+            pool_type_id: self.pool_type_id,
+        }
+    }
+}
+
+pub async fn connect(fqdn: Fqdn) -> Result<PoolClient, Error> {
+    let (client, conn) = iml_postgres::connect().await?;
+
+    tokio::spawn(async move {
+        conn.await
+            .unwrap_or_else(|e| tracing::error!("DB connection error {}", e));
+    });
+
+    let query = format!(
+        "SELECT id FROM django_content_type WHERE app_label = 'chroma_core' AND model = $1"
+    );
+    let s = client.prepare(&query).await?;
+    let row = client.query_one(&s, &[&"ostpool"]).await?;
+    let pool_type_id = row.try_get(0)?;
+
+    Ok(PoolClient {
+        client: Arc::new(client),
+        fqdn: fqdn.to_string(),
+        pool_type_id,
+    })
+}
+
+impl PoolClient {
+    /// Assume that if there is a single filesystem with a given name,
+    /// that it's the correct one
```

Comment:
Why could it be not the correct one?

Reply:
If there are multiple MGS nodes, filesystem names are not guaranteed to be unique.  I don't believe there's a way to get the wrong filesystem, but I wanted to document the assumption.